le.go

```diff
@@ -186,31 +170,66 @@ func (logger *Logger) SetPrefix(prefix string) {
 // Write writes a bytes array to the Logentries TCP connection,
 // it adds the access token and prefix and also replaces
 // line breaks with the unicode \u2028 character
-func (logger *Logger) Write(p []byte) (n int, err error) {
-	if err := logger.ensureOpenConnection(); err != nil {
-		return 0, err
-	}
+func (logger *Logger) Write(p []byte) (int, error) {
+	fmt.Println("le_go: Write()")
 
-	logger.mu.Lock()
-	defer logger.mu.Unlock()
+	buf := logger.makeBuf(p)
 
-	logger.makeBuf(p)
+	n, err := logger.conn.Write(buf)
+	if err == nil {
+		return n, err
+	}
+	fmt.Printf("le_go: Write n=%d, err=%s", n, err.Error())
 
-	return logger.conn.Write(logger.buf)
+	// First write failed.  Try reconnecting and then a second write; if that fails give up.  If
```

Comment:
But these writes are triggered by logger prints scattered everywhere in the code.  Imagine every time you wanted to print some information, you had to attempt a reconnection.